File: experiment-5/runs/language=java_model=claude-opus-4-7_tooling=none/rep1/src/test/java/com/brsoccer/mcp/DataLoadingTest.java
package com.brsoccer.mcp;

import com.brsoccer.mcp.server.SoccerKnowledgeBase;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

@DisplayName("Feature: Data loading")
class DataLoadingTest {

    @Test
    @DisplayName("Given the data directory When the knowledge base loads Then all 6 CSV files are loadable and queryable")
    void allCsvsLoad() {
        SoccerKnowledgeBase kb = TestData.get();
        assertNotNull(kb.matches().all());
        assertNotNull(kb.players().all());
        // 4180 + 1337 + 1255 + 10296 + 6886 = 23,954 matches expected
        assertTrue(kb.matches().all().size() > 20_000,
            "expected >20k matches, got " + kb.matches().all().size());
        assertTrue(kb.players().all().size() > 15_000,
            "expected >15k players, got " + kb.players().all().size());
    }
}

File: experiments/adrianco/experiment-5/runs/language=typescript_model=claude-opus-4-8_tooling=none/rep3-failed/test/dataLoading.test.ts
/**
 * ============================================================================
 * Context Block — File: test/dataLoading.test.ts
 * Project: Brazilian Soccer MCP Server
 * Purpose: BDD specs asserting that all six provided CSV files load and are
 *          queryable (spec success criterion: "All 6 CSV files are loadable and
 *          queryable"), with UTF-8 encoding preserved.
 * ============================================================================
 */

import { describe, it, expect } from 'vitest';
import { givenLoadedDatabase } from './helpers.js';

describe('Feature: Data loading', () => {
  it('Scenario: all six datasets load with non-trivial row counts', () => {
    const db = givenLoadedDatabase();
    const counts = db.sourceCounts;

    // Each provided file contributes records (within a tolerance of the spec's
    // advertised sizes — a few rows may be dropped for missing scores/dates).
    expect(counts['Brasileirao_Matches.csv']).toBeGreaterThan(4000);
    expect(counts['Brazilian_Cup_Matches.csv']).toBeGreaterThan(1200);
    expect(counts['Libertadores_Matches.csv']).toBeGreaterThan(1200);
    expect(counts['BR-Football-Dataset.csv']).toBeGreaterThan(10000);
    expect(counts['novo_campeonato_brasileiro.csv']).toBeGreaterThan(6800);
    expect(counts['fifa_data.csv']).toBeGreaterThan(18000);
  });

  it('Scenario: the knowledge graph exposes matches and players', () => {
    const db = givenLoadedDatabase();
    expect(db.matches.length).toBeGreaterThan(20000);
    expect(db.players.length).toBeGreaterThan(18000);
  });

  it('Scenario: Portuguese accents are preserved (UTF-8)', () => {
    const db = givenLoadedDatabase();
    const accented = db.matches.some(
      (m) => /[ãáàâéêíóôõúç]/i.test(m.homeTeam) || /[ãáàâéêíóôõúç]/i.test(m.awayTeam),
    );
    expect(accented).toBe(true);
  });

  it('Scenario: a season range spanning the historical and modern data', () => {
    const db = givenLoadedDatabase();
    const [first, last] = db.summary().seasonRange;
    expect(first).toBeLessThanOrEqual(2003);
    expect(last).toBeGreaterThanOrEqual(2022);
  });
});
